client/components/Reports/getMetrics.js

```diff
@@ -91,35 +91,44 @@ const countUnexpectedBehaviors = ({
     return countScenarioResult(scenarioResult);
 };
 
-const getMetrics = ({
-    scenarioResult, // Choose one to provide
-    testResult, // Choose one to provide
-    testPlanReport // Choose one to provide
-}) => {
-    const result = { scenarioResult, testResult, testPlanReport };
-    const requiredAssertionsPassedCount = countAssertions({
+const calculateAssertionPriorityCounts = (result, priority) => {
+    const assertionsPassedCount = countAssertions({
         ...result,
-        priority: 'REQUIRED',
+        priority,
         passedOnly: true
     });
-    const requiredAssertionsCount = countAssertions({
+    const assertionsCount = countAssertions({
         ...result,
-        priority: 'REQUIRED'
+        priority
     });
-    const requiredAssertionsFailedCount =
-        requiredAssertionsCount - requiredAssertionsPassedCount;
+    const assertionsFailedCount = assertionsCount - assertionsPassedCount;
 
-    const optionalAssertionsPassedCount = countAssertions({
-        ...result,
-        priority: 'OPTIONAL',
-        passedOnly: true
-    });
-    const optionalAssertionsCount = countAssertions({
-        ...result,
-        priority: 'OPTIONAL'
-    });
-    const optionalAssertionsFailedCount =
-        optionalAssertionsCount - optionalAssertionsPassedCount;
+    return { assertionsCount, assertionsPassedCount, assertionsFailedCount };
+};
+
+const getMetrics = ({
+    scenarioResult, // Choose one to provide
+    testResult, // Choose one to provide
+    testPlanReport // Choose one to provide
+}) => {
+    const result = { scenarioResult, testResult, testPlanReport };
+    const {
+        assertionsCount: requiredAssertionsCount,
+        assertionsPassedCount: requiredAssertionsPassedCount,
+        assertionsFailedCount: requiredAssertionsFailedCount
+    } = calculateAssertionPriorityCounts(result, 'REQUIRED');
```

Comment:
For reviewers: A follow up task to replace instances of `REQUIRED` and `OPTIONAL` could follow this PR landing for consistency. I'd rather not inflate this PR further.